nomos-services/chain-service/src/storage/adapters/storage.rs

```diff
@@ -106,3 +106,17 @@ where
         Ok(Some(deserialized_block))
     }
 }
+
+impl<Storage, Tx, BlobCertificate, RuntimeServiceId> StorageAdapterExt<RuntimeServiceId>
+    for StorageAdapter<Storage, Tx, BlobCertificate, RuntimeServiceId>
+where
+    Storage: StorageBackend + Send + Sync + 'static,
+    <Storage as StorageChainApi>::Block:
+        TryFrom<Block<Tx, BlobCertificate>> + TryInto<Block<Tx, BlobCertificate>>,
+    Tx: Clone + Eq + Serialize + DeserializeOwned + Send + Sync + 'static,
+    BlobCertificate: Clone + Eq + Serialize + DeserializeOwned + Send + Sync + 'static,
+{
+    fn parent_id(block: &Self::Block) -> HeaderId {
+        block.header().parent()
+    }
+}
```

Comment:
uhm, is this really needed? Why can't whatever not be able to access the parent?